### PR TITLE
Fix timer task visibility timestamp for workflow refresh

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Fix `DeleteHistoryEventTask` and `WorkflowBackoffTimerTask`'s visibility timestamp when they are generated by workflow refresh

**Why?**
- Workflow refresh will use the refresh time instead of workflow close/start time to generated those two tasks.

**How did you test it?**
- Tried workflow refresh locally and verified time for generated DeleteHistoryEventTask and WorkflowBackoffTimerTask

**Potential risks**

**Is hotfix candidate?**
- Maybe

### CODE REVIEW

File: service/history/workflow/task_refresher.go [hunk 1]
```diff
@@ -217,8 +216,14 @@ func (r *TaskRefresherImpl) refreshTasksForWorkflowClose(
 	executionState := mutableState.GetExecutionState()
 
 	if executionState.Status != enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING {
+		closeEvent, err := mutableState.GetCompletionEvent(ctx)
```
Comment (yux0) [inline service/history/workflow/task_refresher.go:219]: You can call GetWorkflowCloseTime() to get the close time

File: service/history/workflow/task_generator.go [hunk 2]
```diff
@@ -161,6 +165,7 @@ func (r *TaskGeneratorImpl) GenerateWorkflowStartTasks(
 
 func (r *TaskGeneratorImpl) GenerateWorkflowCloseTasks(
 	now time.Time,
+	closeEvent *historypb.HistoryEvent,
```
Comment (yux0) [inline service/history/workflow/task_generator.go:168]: You can call mutableState.GetWorkflowCloseTime() and pass in the close time

Comment (yycptt) [reply service/history/workflow/task_generator.go:168]: Yeah that's my first attempt on the implementation. However `GetWorkflowCloseTime` requires a context parameter, which means I still need to modify the interface to take in a context value, but currently none of the methods in task generator has context parameter. 

Only task refresh command needs to get the close event from DB, which I think is ok. In all other places, close event is already available when calling this method, so pass in the close event won't have any extra cost.

File: service/history/workflow/task_refresher.go (hunk 1 above)
Comment (yycptt) [reply service/history/workflow/task_refresher.go:219]: Plz check my response to the other comments.

File: service/history/workflow/task_generator.go (hunk 2 above)
Comment (yycptt) [reply service/history/workflow/task_generator.go:168]: I can change it back to use `GetWorkflowCloseTime` or use executionInfo.CloseTime directly when we stop supporting getting close time from close event. I can add a TODO comment here.

**Review by yux0 — APPROVED** [review]

File: service/history/workflow/task_generator.go (hunk 2 above)
Comment (yycptt) [reply service/history/workflow/task_generator.go:168]: Actually I already added a TODO for that on L52

**Pull request merged**